main.go

```diff
@@ -202,3 +215,37 @@ func wrapAuthHandler(handler http.HandlerFunc, htpasswdFile string, host string)
 	authenticator := auth.NewBasicAuthenticator(host, secrets)
 	return auth.JustCheck(authenticator, handler)
 }
+
+func writePidFile(c *config.Config, accessLogger cache.Logger) error {
```

Comment:
What if two bazel-remote daemon start at the same time, and find there is no existing pid file at the same time? It could cause two running daemons of the same version. I think we'd better have a file lock at the beginning of the function to prevent the race condition.

Also checkRunningProcess might be a function name.

Reply:
as long as we specify a port, this won't happen.

Reply:
I don't see how specifying a port can help in this case.

Firstly, two daemons can use different ports.

Even if you specify the same port for both, one daemon cannot start because port binding will fail. But that happens after writing the pid files. Regardless you using one pid file or multiple pid files, they have been messed up.

Reply:
what race condition can it cause? If two bazel-remote start at the same time, they will have different pid, and write to different files. no?

Reply:
The whole purpose of this PR is to use pid file to prevent two bazel-remote process of the same version running. Now you allow two process to run, I am confused what you want to do.